File: server/php/src/Application/DTOs/PaymentLink/ResponseDto.php
<?php

namespace OnlinePayments\ExampleApp\Application\DTOs\PaymentLink;

use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Currency;
use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Status;

class ResponseDto
{
    public ?string $redirectUrl;
    public ?string $paymentLinkId;
    public ?Status $status;
    public ?int $amount;
    public ?Currency $currency;

    public function __construct(
        ?string $redirectUrl,
        ?string $paymentLinkId,
        ?Status $status,
        ?int $amount,
        ?Currency $currency
    ) {
        $this->redirectUrl = $redirectUrl;
        $this->paymentLinkId = $paymentLinkId;
        $this->status = $status;
        $this->amount = $amount;
        $this->currency = $currency;
    }
}

File: server/ruby/app/business/dtos/payment_link/response_dto.rb
module Business
  module Dtos
    module PaymentLink
      class ResponseDto
        attr_accessor :redirect_url, :payment_link_id, :status, :amount, :currency

        def initialize(redirect_url: nil, payment_link_id: nil, status: nil, amount: nil, currency: nil)
          @redirect_url     = redirect_url
          @payment_link_id  = payment_link_id
          @status           = status
          @amount           = amount
          @currency         = currency
        end
      end
    end
  end
end